timeless/reservations/views.py

```diff
@@ -78,6 +80,25 @@ def list_reservations(reservations):
         "restaurants/tables/list.html", reservations=reservations
     )
 
+@bp.route("/retrieve", methods=("GET", "POST"))
+def retrieve_reservations():
```

Comment:
@carlosmiranda according to task so far I can see that endpoint should be linked to a specific company, then we should provide `company_id` to view.